src/Microsoft.SqlTools.ServiceLayer/Scripting/ScriptAsScriptingOperation.cs

```diff
@@ -41,28 +48,26 @@ public override void Execute()
                 this.CancellationToken.ThrowIfCancellationRequested();
                 string resultScript = string.Empty;
                 // TODO: try to use one of the existing connections
-                using (SqlConnection sqlConnection = new SqlConnection(this.Parameters.ConnectionString))
-                {
-                    sqlConnection.Open();
-                    ServerConnection serverConnection = new ServerConnection(sqlConnection);
-                    Server server = new Server(serverConnection);
-                    scripter = new SqlServer.Management.Smo.Scripter(server);
-                    ScriptingOptions options = new ScriptingOptions();
-                    SetScriptBehavior(options);
-                    PopulateAdvancedScriptOptions(this.Parameters.ScriptOptions, options);
-                    options.WithDependencies = false;
-                    options.ScriptData = false;
-                    SetScriptingOptions(options);
-
-                    // TODO: Not including the header by default. We have to get this option from client
-                    options.IncludeHeaders = false;
-                    scripter.Options = options;
-                    scripter.Options.ScriptData = false;
-                    scripter.ScriptingError += ScripterScriptingError;
-                    UrnCollection urns = CreateUrns(serverConnection);
-                    var result = scripter.Script(urns);
-                    resultScript = GetScript(options, result);
-                }
+
+                Server server = new Server(ServerConnection);
```

Comment:
What happens if the connection is closed? Will this handle auto-reopening it or do we need to cover that ourselves? We have cases where the connection is closed and in query execution we now auto reopen before doing the operation